auto_dev/mech/create_mech_tool.py

```diff
@@ -0,0 +1,320 @@
+        try:
+            response = engine.generate_content(
+                prompt,
+                generation_config=genai.types.GenerationConfig(
+                    candidate_count=candidate_count,
+                    stop_sequences=stop_sequences,
+                    max_output_tokens=max_output_tokens,
+                    temperature=temperature,
+                ),
+            )
+            # Ensure response has a .text attribute
+            response_text = getattr(response, "text", None)
+        except Exception as e:
+            return f"An error occurred: {{str(e)}}", None, None, None
+        return response.text, prompt, None, counter_callback
+        ....Edit this to work for the code I about to give you for {tool_name} based on the documentation and only give the code.
+        Output only code, no words. This is being put directly in a Python file. Do not put the coding quotation formatting for python files.
+        Also give me a commented out main function to run this code at the bottom of the file for testing.
+        .....
+        {api_logic_content}
+    """
+
+
+    # Call GPT to generate the content
+    try:
+        response = client.chat.completions.create(
+            model="gpt-4o",
+            messages=[
+                {"role": "user", "content": prompt}
+            ]
+        )
+
+        # Extract GPT's response
+        gpt_response = response.choices[0].message.content.strip()
+    except Exception as e:
+        print(f"Error calling GPT: {e}")
+        sys.exit(1)
+
+    # Write the generated content into the <tool_name>.py file
+    tool_py_path = os.path.join(tool_folder_path, f"{tool_name}.py")
+    try:
+        with open(tool_py_path, 'w') as f:
+            f.write(gpt_response)
+        print(f"Generated content written to {tool_py_path}")
+        return True
+    except Exception as e:
+        print(f"Error writing to {tool_py_path}: {e}")
+        sys.exit(1)
+
+def append_comments_to_tool_file(tool_file_path, comments):
+    """
+    Appends comments to the bottom of the specified tool file.
+    Args:
+        tool_file_path (str): The path to the tool file.
+        comments (str): The comments to append.
+    Returns:
+        None
+    """
+    try:
+        with open(tool_file_path, 'a') as f:
+            f.write("\n\n# " + "\n# ".join(comments.splitlines()))
+        print(f"Comments successfully appended to {tool_file_path}")
+    except Exception as e:
+        print(f"Error appending comments to {tool_file_path}: {e}")
+
+def main():
+    #TODO: Input your GPT KEY HERE
+    parser = argparse.ArgumentParser(description="CLI tool to create a custom Mech tool")
+    parser.add_argument("gpt_key", help="Your OpenAi API Key")
+    parser.add_argument("api_file", help="Python file implementing the API logic")
+    parser.add_argument("tool_name", help="The name for the new tool")
+    parser.add_argument("author_name", help="The name of the author")
```

Comment:
USe click, such that it gets added to the cli tool automatically, see the first few lines of any command group